Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -186,23 +188,21 @@ const PlatformTimeRanges& MediaSource::buffered() const
     return m_buffered;
 }
 
-void MediaSource::waitForTarget(const SeekTarget& target, CompletionHandler<void(const MediaTime&)>&& completionHandler)
+Ref<MediaSource::MediaTimePromise> MediaSource::waitForTarget(const SeekTarget& target)
 {
-    if (isClosed()) {
-        completionHandler(MediaTime::invalidTime());
-        return;
-    }
+    if (isClosed())
+        return MediaTimePromise::createAndReject(-1);
```

Comment:
Should we define a constant instead?

Reply:
I'm replacing this with a constant in part 4 with a PlatformMediaError